### PR TITLE
Make showing and hiding SelfUser.vue cleaner

### PR DESCRIPTION
**Pull request by ItsOnlyBinary**
#391

### CODE REVIEW

File: src/components/ControlInput.vue [hunk 1]
```diff
@@ -399,6 +399,10 @@ export default {
     box-sizing: border-box;
 }
 
+.kiwi-controlinput .kiwi-controlinput-inner i {
```
Comment (prawnsalad) [inline src/components/ControlInput.vue]: This can just be `.kiwi-controlinput-inner i`

File: src/components/SelfUser.vue [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <template>
     <div class="kiwi-selfuser">
+        <div class="close-icon" @click="hide">
```
Comment (prawnsalad) [inline src/components/SelfUser.vue]: `close-icon` should be named correctly by prefixing the component name at the start. Eg. `kiwi-selfuser-close`

File: src/components/SelfUser.vue [hunk 3]
```diff
@@ -31,13 +34,16 @@ export default {
                 str = '+' + str;
             }
 
-            return str;
+            return 'User modes: ' + str;
```
Comment (prawnsalad) [inline src/components/SelfUser.vue]: The 'User modes: ' text isn't needed here as it's obvious what it is already while keeping it less cluttered

File: src/components/SelfUser.vue [hunk 4]
```diff
@@ -31,13 +34,16 @@ export default {
...
         },
         netUser() {
             return this.network.ircClient.user;
         },
     },
     methods: {
+        hide: function hide() {
+            this.parent.selfuser_open = false;
```
Comment (prawnsalad) [inline src/components/SelfUser.vue]: Child components should never interact with its parent directly and passing it in as a prop is just a long about way of doing that. It's better to emit a `close` event and use `@close="selfuser_open=false"` on the parent instead

File: src/components/SelfUser.vue [hunk 5]
```diff
@@ -54,6 +60,19 @@ export default {
...
+    right: 0;
+    top: 0;
+    color: #fff;
+    cursor: pointer;
+    font-size: 0.8em;
+    padding: 0.2em 0.4em;
+    border-radius: 0 0 0 0.4em;
+    transition: all 0.3s;
+    background-color: #fc6262;
+}
```
Comment (prawnsalad) [inline src/components/SelfUser.vue]: Colour related styles should be in the theme files, not in the core

File: src/components/SelfUser.vue (hunk 3 above)
Comment (ItsOnlyBinary) [reply src/components/SelfUser.vue]: im not convinced this is clear how it originally was for me it was just a line with +cx but no explanation of that +cx is

Comment (ItsOnlyBinary) [reply src/components/SelfUser.vue]: image

**Pull request merged**